contracts/Vault.vy

```diff
@@ -1190,50 +1190,22 @@ def expectedReturn(_strategy: address = msg.sender) -> uint256:
     return self._expectedReturn(_strategy)
 
 
-@external
-def report(_gain: uint256, _loss: uint256) -> uint256:
-    """
-    @notice
-        Reports the amount of assets the calling Strategy has free (usually in
-        terms of ROI).
-
-        This may only be called by a Strategy managed by this Vault.
-    @dev
-        For approved strategies, this is the most efficient behavior.
-        The Strategy reports back what it has free, then Vault "decides"
-        whether to take some back or give it more. Note that the most it can
-        take is `_gain`, and the most it can give is all of the remaining
-        reserves. Anything outside of those bounds is abnormal behavior.
-
-        All approved strategies must have increased diligence around
-        calling this function, as abnormal behavior could become catastrophic.
-    @param _gain
-        Amount Strategy has realized as a gain on it's investment since its
-        last report, and is free to be given back to Vault as earnings
-    @param _loss
-        Amount Strategy has realized as a loss on it's investment since its
-        last report, and should be accounted for on the Vault's balance sheet
-    @return Amount of debt outstanding (iff totalDebt > debtLimit).
-    """
-
-    # Only approved strategies can call this function
-    assert self.strategies[msg.sender].activation > 0
+@internal
+def _reportLoss(_strategy: address, _loss: uint256):
+    # Loss can only be up the amount of debt issued to strategy
+    totalDebt: uint256 = self.strategies[_strategy].totalDebt
+    loss: uint256 = min(_loss, totalDebt)
+    self.strategies[_strategy].totalLoss += loss
+    self.strategies[_strategy].totalDebt = totalDebt - loss
+    self.totalDebt -= loss
 
-    # We have a loss to report, do it before the rest of the calculations
-    if _loss > 0:
-        loss: uint256 = min(_loss, self.strategies[msg.sender].totalDebt)
-        self.strategies[msg.sender].totalLoss += loss
-        self.strategies[msg.sender].totalDebt -= loss
-        self.totalDebt -= loss
-        # Also, make sure we reduce our trust with the strategy by the same amount
-        self.strategies[msg.sender].debtLimit -= min(
-            loss,
-            self.strategies[msg.sender].debtLimit,
-        )
+    # Also, make sure we reduce our trust with the strategy by the same amount
+    debtLimit: uint256 = self.strategies[_strategy].debtLimit
+    self.strategies[_strategy].debtLimit -= min(loss, debtLimit)
```

Comment:
is this `min` necessary?

Reply:
Just in case the strategy reports loss incorrectly, we want to protect against an underflow here